at-quannguyen/W4/main.rb

```diff
@@ -0,0 +1,621 @@
+
+    user_input = @session_controller.username_by_employee username
+    role = user_input.to_a[0]['role']
+    team_id = user_input.to_a[0]['team_id']
+
+    if user_input.count.zero?
+      puts 'Not find employee !'
+      csleep
+      cexit
+      action_add_member_team
+    end
+
+    unless team_id.nil?
+      puts 'User there are teams !'
+      csleep
+      cexit
+      action_add_member_team
+    end
+
+    if role.zero?
+      if @team_controller.add_member(team_current_id, username)
+        puts 'Add complete, you want to add more!'
+        csleep
+        cexit
+        action_add_member_team
+      end
+    else
+      puts 'No add leader end system!'
+      csleep
+      cexit
+      action_add_member_team
+    end
+  end
+
+  def action_show_members_team
+    clear
+    puts '========= Show All Members Team ========='
+    print 'Name team : '
+    name_team = gets.chomp
+    query = @team_controller.select_obj_by_team name_team
+    team_id = query.to_a[0]['id']
+    members = @team_controller.members(team_id)
+    puts 'ID|USERNAME'
+    members.each do |e|
+      puts e['id'].to_s + '|' + e['username']
+    end
+  end
+
+  def action_update_team
+    clear
+    puts '========= Update Team ========='
+    puts '1. update name'
+    puts '2. update team leader'
+    print 'Action : '
+    number = begin
+               Integer(gets.chomp)
+             rescue StandardError
+               false
+             end
+    if number == false
+      puts 'input not string !'
+      csleep
+      cexit
+      action_update_team
+    end
+    action_update_team_name if number == 1
+    action_update_team_leader if number == 2
+  end
+
+  def action_update_team_name
+    clear
+    puts '========= Update Name Team ========='
+    print 'name team : '
+    name_team = gets.chomp
+    puts ''
+    print 'name team replace : '
+    name_replace = gets.chomp
+
+    team = @team_controller.select_obj_by_team name_team
+
+    if team.count.zero?
+      puts 'not find team !'
+      csleep
+      cexit
+      action_update_team_name
+    end
+
+    if @team_controller.update_name(name_replace, team.to_a[0]['id'])
+      puts 'update complete !'
+      csleep
+      action_team_manager
+    else
+      puts 'update fail, you try !'
+      csleep
+      cexit
+      action_update_team_name
+    end
+  end
+
+  def action_update_team_leader
+    clear
+    puts '========= Update Leader Team ========='
+    print 'name team : '
+    name_team = gets.chomp
+    puts ''
+    print 'username up leader : '
+    user_up_leader = gets.chomp
+
+    team = @team_controller.select_obj_by_team name_team
+    leader_up = @session_controller.username_by_employee user_up_leader
+
+    if team.count.zero? || leader_up.count.zero?
+      puts 'not find team end not add leader !'
+      csleep
+      cexit
+      action_update_team_leader
+    end
+    @team_controller.update_leader(team, leader_up)
+  end
+
+  def action_notification_manager
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/7deb92b5e1c7275222ca44a289eebcc27cbc38fe/at-quannguyen/W4/main.rb#L366"></span>Metrics/PerceivedComplexity: Perceived complexity for action_notification_manager is too high. [10/8]</td>
    </tr>
  </tbody>
</table>